frontend/src/js/previous-queries/list/ProjectItemDragContainer.tsx

```diff
@@ -0,0 +1,74 @@
+import { FC, useRef } from "react";
+import { useDrag } from "react-dnd";
+
+import { getWidthAndHeight } from "../../app/DndProvider";
+import { DNDType } from "../../common/constants/dndTypes";
+import type { DragItemFormConfig } from "../../external-forms/types";
+import type { DragItemQuery } from "../../standard-query-editor/types";
+
+import ProjectItem, { ProjectItemT } from "./ProjectItem";
+import { isFormConfig } from "./helpers";
+import { PreviousQueryT } from "./reducer";
+
+interface PropsT {
+  item: ProjectItemT;
+  onIndicateDeletion: () => void;
+  onIndicateShare: () => void;
+  onIndicateEditFolders: () => void;
+}
+
+const getDragType = (item: PreviousQueryT) => {
+  return item.queryType === "CONCEPT_QUERY"
+    ? DNDType.PREVIOUS_QUERY
+    : DNDType.PREVIOUS_SECONDARY_ID_QUERY;
+};
+
+const ProjectItemDragContainer: FC<PropsT> = ({ item, ...props }) => {
+  const ref = useRef<HTMLDivElement | null>(null);
+
+  const dragItemBase = {
+    dragContext: {
+      width: 0,
+      height: 0,
+    },
+    id: item.id,
+    label: item.label,
+    tags: item.tags,
+    own: item.own,
+    shared: item.shared,
+  };
+  const dragItem: DragItemQuery | DragItemFormConfig = isFormConfig(item)
```

Comment:
Du könntest die zwei branches vereinigen indem du FORM_CONFIG in getDragType ziehst und canExpand nachlagerst an isFormConfig gebunden.

Reply:
Yeah ... but then getDragType gets more complicated to type, so I went for this way